plugins/module_utils/utils.py

```diff
@@ -26,3 +28,173 @@ def generate_api_endpoint(path, **kwargs):
     # return full_url
 
     return path if not kwargs else "{0}?{1}".format(path, "&".join(["{0}={1}".format(key, value) for key, value in kwargs.items()]))
+
+
+def append_update_ops_data(ops, existing_data, update_path, changes=dict(), op="replace"):
+    """
+    Append Update ops payload data.
+    :param ops: Variable which contains the PATCH replace actions for the update operation -> List
+    :param existing_data: Variable which contains the existing data -> Dict
+    :param update_path: The object path is used to update an existing object -> Str
+    :param changes: Input dictionary which contains the attribute to be updated and its new value -> Dict
+    :param op: Defaults to "replace" when not specified, value "remove" is used clear the existing configuration -> Str
+    :return: None
+                If attributes is not empty then the ops and existing_data are updated with the input value.
+
+    Sample data for the replace call:
+    ---------------------------------
+
+    Sample Existing Data:
+    ---------------------
+    existing_data = {
+        "bfdMultiHopPol": {"adminState": "enabled", "minRxInterval": 250},
+        "bfdPol": {"adminState": "enabled", "detectionMultiplier": 3},
+        "name": "irp_1",
+    }
+    ops = []
+    update_path = "/tenantPolicyTemplate/template/l3OutIntfPolGroups/0"
+
+    replace_data = {
+        ("name"): "new_name",
+        "description": "new_description",
+        ("bfdMultiHopPol", "ifControl"): dict(),
+        ("bfdMultiHopPol", "ifControl", "adminState"): "enabled",
```

Comment:
Is there a scenario where it's interesting to put an empty dict? otherwise, we should merge these two examples into one:
```suggestion
        ("bfdMultiHopPol", "ifControl"): dict(adminState="enabled"),
```

Reply:
The current code supports merging the two lines into single-line logic.